CMakeLists.txt

```diff
@@ -514,74 +282,12 @@ write_basic_package_version_file(
-  "${INSTALL_CAF_CMAKEDIR}")
-
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/CAFConfig.cmake"
-  "${CMAKE_CURRENT_BINARY_DIR}/CAFConfigVersion.cmake"
-  DESTINATION "${INSTALL_CAF_CMAKEDIR}")
-
-################################################################################
-#                     Add additional project files to GUI                      #
-################################################################################
-
-file(GLOB_RECURSE script_files "scripts/*")
-add_custom_target(gui_dummy SOURCES configure ${script_files})
-
-################################################################################
-#                                print summary                                 #
-################################################################################
-
-# little helper macro to invert a boolean
-macro(invertYesNo in out)
-  if(${in})
-    set(${out} no)
-  else()
-    set(${out} yes)
-  endif()
-endmacro()
-# invert CAF_NO_* variables for nicer output
-invertYesNo(CAF_NO_IO CAF_BUILD_IO)
-invertYesNo(CAF_NO_EXAMPLES CAF_BUILD_EXAMPLES)
-invertYesNo(CAF_NO_TOOLS CAF_BUILD_TOOLS)
-invertYesNo(CAF_NO_UNIT_TESTS CAF_BUILD_UNIT_TESTS)
-invertYesNo(CAF_NO_EXCEPTIONS CAF_BUILD_WITH_EXCEPTIONS)
-invertYesNo(CAF_NO_MEM_MANAGEMENT CAF_BUILD_MEM_MANAGEMENT)
-invertYesNo(CAF_NO_OPENSSL CAF_BUILD_OPENSSL)
-if(GENERATOR_IS_MULTI_CONFIG)
-  set(summary_build_type "Multi-Config")
-else()
-  set(summary_build_type "${CMAKE_BUILD_TYPE}")
-endif()
-# done
-if(NOT CAF_NO_SUMMARY)
-  message(STATUS
-        "\n====================|  Build Summary  |===================="
-        "\n"
-        "\nCAF version:           ${CAF_VERSION}"
-        "\n"
-        "\nCXX:                   ${CMAKE_CXX_COMPILER}"
-        "\nBuild type:            ${summary_build_type}"
-        "\nLink type:             ${LINK_TYPE}"
-        "\nBuild static runtime:  ${CAF_BUILD_STATIC_RUNTIME}"
-        "\nRuntime checks:        ${CAF_ENABLE_RUNTIME_CHECKS}"
-        "\nLog level:             ${CAF_LOG_LEVEL}"
-        "\nWith mem. mgmt.:       ${CAF_BUILD_MEM_MANAGEMENT}"
-        "\nWith exceptions:       ${CAF_BUILD_WITH_EXCEPTIONS}"
-        "\n"
-        "\nBuild I/O module:      ${CAF_BUILD_IO}"
-        "\nBuild OpenSSL module:  ${CAF_BUILD_OPENSSL}"
-        "\nBuild tools:           ${CAF_BUILD_TOOLS}"
-        "\nBuild examples:        ${CAF_BUILD_EXAMPLES}"
-        "\nBuild unit tests:      ${CAF_BUILD_UNIT_TESTS}"
-        "\n"
-        "\n"
-        "\nSource directory:      ${CMAKE_CURRENT_SOURCE_DIR}"
-        "\nBuild directory:       ${CMAKE_CURRENT_BINARY_DIR}"
-        "\nExecutable path:       ${EXECUTABLE_OUTPUT_PATH}"
-        "\nLibrary path:          ${LIBRARY_OUTPUT_PATH}"
-        "\nInstall prefix:        ${CMAKE_INSTALL_PREFIX}"
-        "\nGenerator:             ${CMAKE_GENERATOR}"
-        "\n"
-        "\n===========================================================\n")
```

Comment:
I liked the summary, just to check flags and stuff.

Reply:
Those are available in the CMake cache and/or compile database. Including flags here is borderline useless, because they depend on the configuration when using a multi-configuration generator.